osf-cli/README.md

```diff
@@ -0,0 +1,83 @@
+
+##How It Works
+The CLI uses functionality in the `osf-packager` module and the OSF Java client in the `osf-client` package to attach to the OSF REST APIs (both v1 and v2) on a running OSF instance specified by a configuration file whose location must be supplied by the user. Additional parameters will need to be supplied as indicated below. The packager code leverages a workflow from the Dataconservancy Package Tool GUI to construct the package once the content from the target registration has been retrieved. The package is then saved to a location specified by the user as a command line option.
+
+
+##Command Line Usage
+The CLI takes a single argument which is a URL pointing to the registration whcih is to be packaged (for example, `http://192.168.99.100:8000/v2/registrations/vacbu/`
+This is preceded by a list of command line options as follows:
+
+```
+-c (-configuration, --configuration)  FILE   : path to the OSF Java client configuration
+-h (-help, --help)                           : print help message
+-m (-metadata, --metadata) FILE              : the path to the metadata properties file for additional bag metadata
+-n (-name, --name) VAL                       : the name for the bag
+-o (-output, --output) FILE                  : path to the directory where the package will be written
+-v (-version, --version)                     : print version information
+```
+The `-c, -m, -n` and `-o` options are required. The OSF Java client must be configured so that it knows which running OSF instance to attach to. Additional metadata is specified in a bag metadata properties file, and is used to describe the bag in accordance with the Data Conservancy BagIt Profile. The package name is used to both name the package and the bag. Finally, the output location will tell the CLI where to write the package.
+
+###OSF Java Client Configuration
+Configuration mut be supplied for both the v1 and v2 endpoints, since both are needed to build the package.  An example is below:
+```
+{
+  "osf": {
+    "v2": {
+      "host": "192.168.99.100",
+      "port": "8000",
+      "basePath": "/v2/",
+      "authHeader": "Basic ZW1ldHNnZXJAZ21haWwuY29tOmZvb2JhcmJheg==",
+      "scheme": "http"
+    }
+  },
+  "wb": {
+    "v1": {
+      "host": "192.168.99.100",
+      "port": "7777",
+      "basePath": "/v1/",
+      "scheme": "http"
+    }
+  }
+}
+```
+The absolute path to the file containing this configuration is what must be supplied as the value for the `-c` option.
+
+###Bag Metadata File
+The `-m` option also takes an absolute path as its value. This file contains bag metadata which will conforn to the Data Conservancy BagIt Profile specification. Metadata entries will consist of key = value pairs, one per line. The values may be lists where appropriate, expressed as comma separated strings. If the value of Package-Name is specified in this file, it will be superseded by the value specified for the `-n` option below. 
```

Comment:
Yeah, so this is interesting - I believe that we are generating the only 
truly required bag metadata according to the spec - but we have a 
different notion of "required" in the GUI, a lot more stuff is enforced 
than the specification's minimum (contact info for example).

i think that if a minimally conforming bag is the goal, then we may not 
need to require the bag metadata file at all. if compatibility with the 
command line tool is desired, then we may have to look at all the 
required properties and spell them all out. Your thoughts?

Jim

On 08/02/2016 03:27 PM, Elliot Metsger wrote:
>
> In osf-cli/README.md 
> <https://github.com/emetsger/osf-packaging/pull/6#discussion_r73220826>:
>
> > +    }
> > +  },
> > +  "wb": {
> > +    "v1": {
> > +      "host": "192.168.99.100",
> > +      "port": "7777",
> > +      "basePath": "/v1/",
> > +      "scheme": "http"
> > +    }
> > +  }
> > +}
> > +```
> > +The absolute path to the file containing this configuration is what must be supplied as the value for the `-c` option.
> > +
> > +###Bag Metadata File
> > +The `-m` option also takes an absolute path as its value. This file contains bag metadata which will conforn to the Data Conservancy BagIt Profile specification. Metadata entries will consist of key = value pairs, one per line. The values may be lists where appropriate, expressed as comma separated strings. If the value of Package-Name is specified in this file, it will be superseded by the value specified for the `-n` option below.
>
> Do we have any online documentation we can link to here for supported 
> Bag metadata properties? The docs say this command line switch is 
> required, so if that is actually the case, it would be helpful to have 
> a list of the property keys and example values that are required.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/emetsger/osf-packaging/pull/6/files/ebccbed8c1362bb6545e2fd644f9e5f50e3bf929#r73220826>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AAS-tyIMwBfwpEFICNb79haeMHAbPxqSks5qb5oYgaJpZM4JXnbx>.
>